src/qt/macdockiconhandler.mm

```diff
@@ -11,52 +11,46 @@
 
 #undef slots
 #include <Cocoa/Cocoa.h>
+#include <objc/objc.h>
+#include <objc/message.h>
 
 #if QT_VERSION < 0x050000
 extern void qt_mac_set_dock_menu(QMenu *);
 #endif
 
-@interface DockIconClickEventHandler : NSObject
-{
-    MacDockIconHandler* dockIconHandler;
-}
-
-@end
-
-@implementation DockIconClickEventHandler
+static MacDockIconHandler *s_instance = NULL;
```

Comment:
I'd prefer if it was freed. Otherwise it will show up as a memory leak. Would be even better to do it without introducing a singleton, but I suppose that isn't possible.